### ISSUE TITLE
Color swatches do not update after using replace command.

### ISSUE THREAD

**Issue by Jose-Moreno** [pencil2d/pencil#1055]
### --Issue Summary--
Ok so when using vector layers when you change a color it's value updates but the thumbnail doesn't. se Using the "replace" command updates the swatch color value and thumbnail but not it's name.

However when a bitmap layer is active the palette behaves a bit differently. 

You can change the color locally, and while the value changesthe thumbnail doesn't update, but once you select a different swatch the value of the swatch itself returns to a previous value. Using the replace command does not update the swatch color value, its name, but it does update its thumbnail.

Related to #943 using the replace command on the first swatch of the palette modifies the color but not the thumbnail in vector, but doesn't change anything for bitmap layers.

Summary:
Vector: 
- Replace command updates:
    - [x] Color
    - [x] Thumbnail (Not the first swatch)
    - [ ] Generated Name
Bitmap: 
- Replace command updates:
    - [ ] Color
    - [ ] Thumbnail
    - [ ] Generated Name

Discussion:
It seems not updating the swatch thumbnail when modifying a color with the wheel/ inspector was implemented recently and is intended to avoid unwanted changes to the color, however I've noticed that the vector layer does not respect the "replace" command philosophy as you can permanently change the color without using such command. Ideally this functionality should be consistent on both layer types.

### --Expected Results--
The replace command should properly update all the noted properties and values (the generated name is debatable but still..) and its use should be consistent on either layer type.

### --Video or Image Reference--
Will record a video soon.

### --Steps to reproduce--
Will update later.

### --System Information--
+ Pencil2D Version:
Version: 0.6.1.1
commit: 0a2c87b523fc81c655daf391a26af967134ba733
date: 2018-08-20_11:39:21
Development build
Operating System: Windows 7 SP 1 (6.1)
CPU Architecture: x86_64

**Comment by chchwy** [comment]
The first colour of palette never updating has been fixed.

**Comment by Jose-Moreno** [comment]
_**Update 03/06/2019**_

With 0.6.4 changing the color on the vector layer in real-time (color widgets), no longer updates the swatches until the display mode is changed between either list or grid. Using the replace command does change the thumbnail.

Am i to understand that we are going to suggest our users not to update their vector palette in real time, pretty much like the bitmap palette, instead forcing them to replace the color? If so, that's fine, but then we should unify the behavior for both palette types.

Related to #943 All the swatches along with the first refuse to update the RGB color when using the replace command

**Comment by Jose-Moreno** [comment]
@Jose-Moreno Reminder to test these issues after #1265 was merged. If any was not fixed I'll open a new issue for that specific problem instead.

**Comment by Jose-Moreno** [comment]
Alright. I've tested this. It's working as expected. Thanks!

I'm still missing the generated name change when replacing a color but I'll add that separately as a wish list feature request instead of part of a bug report.

What I'd take as expected behavior in that case, is that if the name has not been changed, the replace command can invoke the name generator to update the color string, else if the name has been modified in anyway, then it shouldn't run to avoid changing the personalized name (current behavior).
